src/main/java/org/traccar/model/ExtendedModel.java

```diff
@@ -154,4 +154,56 @@ public long getLong(String key) {
         }
     }
 
+    public void removeAttribute(String key) {
+        attributes.remove(key);
+    }
+
+    public String removeString(String key) {
+        var value = attributes.remove(key);
+        return value != null ? value.toString() : null;
+    }
+
+    public Double removeDouble(String key) {
+        var value = attributes.remove(key);
+        if (value == null) {
+            return null;
+        } else if (value instanceof Number numberValue) {
+            return numberValue.doubleValue();
+        } else {
+            return Double.parseDouble(value.toString());
+        }
```

Comment:
Also I think we need to reuse the login between `getXxx` and `removeXxx`. There's too much duplication.

Reply:
It has the baked-in default values behavior, which does not match the current logic of returning nulls when there's no such key, so nothing comes to my mind apart from the extraction of the few common lines to a function, which feels wrong

Reply:
Duplicating the code is never the right solution. I'm sure there's a way to unify it.